### ISSUE TITLE
Windows build server 2008 R2: ReferenceError: Unknown plugin "transform-runtime" specified in "\\node_modules\\regenerator-transform\\package.json" at 0, attempted to resolve relative to "\\node_modules\\regenerator-transform" (While processing preset: "\\node_modules\\babel-preset-react-native\\index.js") 

### ISSUE THREAD

**Issue by kundantmz** [facebook/react-native#15015]
### Is this a bug report?

Not Sure : only facing on Windows Server 2008 
No issue reported while running on Mac OS

### Have you read the Bugs section of the Contributing to React Native Guide?

(Write your answer here.)

### Environment

1. `react-native -v`:0.46.2 || 0.31.0
2. `node -v`:6.11.0
3. `npm -v`:3.10.10
4. `yarn --version` (if you use Yarn):

Then, specify:

- Target Platform: Android

- Development Operating System: Windows Server 2008

- Build tools: Android SDK

### Steps to Reproduce

(Write your steps here:)
Motto : Running Unit Test cases using Mocha 
1. Setup the compiler.js and init.js under test/support folder
2. run the script: mocha --compilers js:./test/support/compiler --require ./test/support/init test/*.test.js test/**/*.test.js

### Expected Behavior

It should execute the unit test cases and produce the report
(Write what you thought would happen.)

### Actual Behavior

node_modules\babel-core\lib\transformation\file\options\option-manager.js:328
        throw e;
        ^
 
ReferenceError: Unknown plugin "transform-runtime" specified in "node_modules\\regenerator-transform\\package.json" at 0, attempted to resolve relative to "node_modules\\regenerator-transform" (While processing preset: "node_modules\\babel-preset-react-native\\index.js") (While processing preset: "node_modules\\babel-preset-react\\lib\\index.js") (While processing preset: "node_modules\\babel-preset-es2015\\lib\\index.js")
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:180:17
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (node_modules\babel-plugin-transform-regenerator\lib\index.js:3:16)
    at Module._compile (module.js:570:32)
    at Module.replacementCompile (node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at require.extensions..js (test\support\compiler.js:41:17)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:178:20
    at Array.map (native)
    at Function.normalisePlugins (node_modules\babel-core\lib\transformation\file\options\option-manager.js:158:20)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:234:36)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:296:17
    at Array.map (native)
    at OptionManager.resolvePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:275:20)
    at OptionManager.mergePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:264:10)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:249:14)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:296:17
    at Array.map (native)
    at OptionManager.resolvePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:275:20)
    at OptionManager.mergePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:264:10)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:249:14)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:296:17
    at Array.map (native)
    at OptionManager.resolvePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:275:20)
    at OptionManager.mergePresets (node_modules\babel-core\lib\transformation\file\options\option-manager.js:264:10)
    at OptionManager.mergeOptions (node_modules\babel-core\lib\transformation\file\options\option-manager.js:249:14)
    at OptionManager.init (node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (node_modules\babel-core\lib\transformation\file\index.js:212:65)
    at new File (node_modules\babel-core\lib\transformation\file\index.js:135:24)
    at Pipeline.transform (node_modules\babel-core\lib\transformation\pipeline.js:46:16)
    at require.extensions..js (test\support\compiler.js:17:18)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at node_modules\mocha\bin\_mocha:355:3
    at Array.forEach (native)
    at Object.<anonymous> (node_modules\mocha\bin\_mocha:354:10)
    at Module._compile (module.js:570:32)
    at Module.replacementCompile (node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at Object.<anonymous> (node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Function.Module.runMain (module.js:604:10)
    at Function.runMain (C:\Users\a-rp79\.node-spawn-wrap-10776-783142caf287\node:40:10)
    at Object.<anonymous> (node_modules\nyc\bin\wrap.js:23:4)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Function.Module.runMain (module.js:604:10)
    at Object.<anonymous> (C:\Users\a-rp79\.node-spawn-wrap-10776-783142caf287\node:140:8)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3
npm ERR! Test failed.  See above for more details.

-----------------------------------------------------------------
upon npm install babel-plugin-transform-runtime
node_modules\babel-core\lib\transformation\file\options\option-manager.js:328
        throw e;
        ^
Error: Couldn't find preset "env" relative to directory "node_modules\\regenerator-transform" (While processing preset: "node_modules\\babel-preset-react-native\\index.js") (While processing preset: "node_modules\\babel-preset-react\\lib\\index.js") (While processing preset: "node_modules\\babel-preset-es2015\\lib\\index.js")
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:293:19

---------------------------------------------------------------------
upon npm install babel-preset-env
 
node_modules\babel-core\lib\transformation\file\options\option-manager.js:328
        throw e;
        ^
Error: Options {"loose":true} passed to node_modules\babel-preset-env\lib\index.js which does not accept options. (While processing preset: "node_modules\\babel-preset-env\\lib\\index.js") (While processing preset: "node_modules\\babel-preset-env\\lib\\index.js") (While processing preset: "node_modules\\babel-preset-react-native\\index.js") (While processing preset: "node_modules\\babel-preset-react\\lib\\index.js") (While processing preset: "node_modules\\babel-preset-es2015\\lib\\index.js")
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:314:17
 
-------------------------------------------------------------------------
upon removing .babelrc throws
node_modules\babel-core\lib\transformation\file\options\option-manager.js:180
          throw new ReferenceError(messages.get("pluginUnknown", plugin, loc, i, dirname));
          ^
 
ReferenceError: Unknown plugin "transform-runtime" specified in "node_modules\\regenerator-transform\\package.json" at 0, attempted to resolve relative to "node_modules\\regenerator-transform"
    at node_modules\babel-core\lib\transformation\file\options\option-manager.js:180:17
 

### Reproducible Demo

On windows Operating System
react-native init helloworld
setup Mocha unit test reference app https://github.com/tigershen23/react-native-redux-mocha-example.git

Snippet of compiler.js
/*eslint-disable */

var fs = require('fs');
var path = require('path');
var babel = require('babel-core');
var origJs = require.extensions['.js'];

require.extensions['.js'] = function (module, fileName) {
  var output;
  if (fileName.match('node_modules/react-native/Libraries/react-native/react-native.js')) {
    fileName = path.resolve('./test/support/mocks/react-native.js');
  }
  if (fileName.indexOf('node_modules/') >= 0) {
    return (origJs || require.extensions['.js'])(module, fileName);
  }
  var src = fs.readFileSync(fileName, 'utf8');
  output = babel.transform(src, {
    filename: fileName,
    sourceFileName: fileName,

    //keep below in sync with babelrc
    "retainLines": true,
    "compact": true,
    "comments": false,
    "presets": ["es2015", "react"],
    "plugins": [
      "syntax-async-functions",
      "syntax-class-properties",
      "syntax-trailing-function-commas",
      "transform-class-properties",
      "transform-flow-strip-types",
      "transform-object-assign",
      "transform-object-rest-spread",
      "transform-react-display-name",
      "transform-react-jsx",
      "transform-regenerator"
    ],
    "sourceMaps": false
  }).code;

  return module._compile(output, fileName);
};

**Comment by hramos** [comment]
Can you fix your code blocks? You can use triple ticks, and also surround long blobs of text with a <detail> tag if needed. It's a bit hard to parse this issue at a glance.

**Comment by kundantmz** [comment]
I couldn't get what you suggested, My problem is like this:
I have a react-native android mobile project, the unit test cases are written using Enzyme and Mocha. The test cases are running fine on Mac machine. However the same test cases are unable to execute on windows machine. I am getting error from babel presets es2015, react and react-native( ReferenceError: Unknown plugin "transform-runtime" specified in "node_modules/regenerator-transform/package.json"). Which i believe some limitations of windows to manage babel node_modules.

**Comment by kundantmz** [comment]
@hramos The problem is around babel-preset transformation on windows platform, any thought on the fixes ?
